test/api/python/provisioner/test_inputs.py

```diff
@@ -744,3 +746,152 @@ def _check(*args):
     # urls
     res = SWUpdateRepo(some_release, source='https://some/http/url')
     _check('is_remote')
+
+
+def test_inputs_copy_attr_verify_attrs():
+    some_cls = attr.make_class("some_cls", {
+        "some_attr": attr.ib(
+            default=123,
+            validator=None,
+            metadata={
+                'metadata_key': 1
+            },
+            repr=False,
+            hash=None,
+            init=True,
+            type=str,
+            converter=None,
+            kw_only=True
+        )
+    })
+
+    read_attr = attr.fields(some_cls).some_attr
+
+    copied_attr = inputs.copy_attr(read_attr)
+
+    assert copied_attr == read_attr
+
+
+def test_ParserFiller_fill_parser_when_action_is_store_bool():
+    parser = argparse.ArgumentParser()
+    SC = attr.make_class("SC", {
+        "y": attr.ib(
+            default=123,
+            metadata={
+                METADATA_ARGPARSER: {
+                    'action': 'store_bool',
+                    'help': 'some help'
+                }
+            },
+            type=int
+        )
+    })
+    ParserFiller.fill_parser(SC, parser)
+    args = parser.parse_args(['--y', '--noy'])
+    assert not args.y
+
+    args = parser.parse_args(['--y'])
+    assert args.y
+
+    args = parser.parse_args(['--noy'])
+    assert not args.y
+
+
+def test_ParserFiller_fill_parser_with_no_action_in_metadata():
+    parser = argparse.ArgumentParser()
+    SC = attr.make_class("SC", {
+        "y": attr.ib(
+            default=123,
+            metadata={
+                METADATA_ARGPARSER: {
+                    'help': 'some help'
+                }
+            },
+            type=int
+        )
+    })
+    ParserFiller.fill_parser(SC, parser)
+    args = parser.parse_args(['--y', 'some-value'])
+    assert args.y == 'some-value'
+
+
+def test_ParserFiller_fill_parser_with_no_metadata_argparser():
+    parser = argparse.ArgumentParser()
+    SC = attr.make_class("SC", {
+        "y": attr.ib(
+            default=123,
+            type=int
+        )
+    })
+    ParserFiller.fill_parser(SC, parser)
+
+
+def test_ParserFiller_fill_parser_input_checks(mocker):
+    parser = argparse.ArgumentParser()
+    SC = attr.make_class("SC", {
+        "y": attr.ib(
+            default=123,
```

Comment:
a boolean is expected here as a default value

Reply:
Done